iphone/Maps/UI/Settings/MWMSettingsViewController.mm

```diff
@@ -191,6 +196,22 @@ - (void)configNavigationSection
     voiceInstructions = L(@"duration_disabled");
   }
   [self.voiceInstructionsCell configWithTitle:L(@"pref_tts_language_title") info:voiceInstructions];
+
+  using namespace routing;
+
+  auto info = [](SpeedCameraManager::Mode mode) {
+    switch (mode)
+    {
+    case SpeedCameraManager::Mode::Auto:
+      return L(@"speedcam_option_auto");
+    case SpeedCameraManager::Mode::Always:
+      return L(@"speedcam_option_always");
+    case SpeedCameraManager::Mode::Never:
+      return L(@"speedcam_option_never");
+    }
+  }(f.GetRoutingManager().GetSpeedCamManager().GetMode());
+
+  [self.speedCamsCell configWithTitle:L(@"speedcams_alert_title") info:info];
```

Comment:
Passing a block here looks too complicated and confusing. Just pass a string.

Reply:
I know, but why don't just
```objective-c
NSString  *info;
switch (f.GetRoutingManager().GetSpeedCamManager().GetMode())
{
case SpeedCameraManager::Mode::Auto:
  info = L(@"speedcam_option_auto");
case SpeedCameraManager::Mode::Always:
  info = L(@"speedcam_option_always");
case SpeedCameraManager::Mode::Never:
  info = L(@"speedcam_option_never");
}
```

Reply:
I like the way I did

Reply:
Is there any reason to have a block rather then a plain `switch` statement?

Reply:
I rewrote the code a day ago the way you told me, please check it again.